src/mixins/ui/lasso-tool-ui.js

```diff
@@ -170,19 +287,52 @@ class ShapeHandler {
       this.canvas.blur()
 
       this.active = false
+
+      this.fireEvent(LassoGlobalEventConstants.LASSO_TOOL_TYPE_DEACTIVATED)
     }
   }
 
-  destroy() {
+  /**
+   * Called under various contexts to destroy either temporary shapes that might have
+   * been created during the shape draw and optionally destroy the primary shape that
+   * may have been drawn
+   * @param {DestroyType} destroy_type Enum describing the context under which destroy()
+   *                                   was called.
+   */
+  // eslint-disable-next-line no-unused-vars
+  destroy(destroy_type) {
     assert(
       false,
       `${ShapeHandler.name}::destroy() needs to be overridden by derived class`
     )
   }
+
+  /**
+   * Called to fire a tool-create-ended event signal, using a destroy type to provide
+   * some extra metadata as to why the draw stopped.
+   * @param {DestroyType} destroy_type Enum describing the context under why the draw stopped
+   */
+  notifyDrawStop(destroy_type) {
+    this.fireEvent(LassoGlobalEventConstants.LASSO_TOOL_DRAW_ENDED, {
+      ended_reason: destroy_type.toString()
```

Comment:
and `ended_reason` property is added to this event signal object as a quick way to describe the reason for the draw end, in case that's useful.